build.gradle

```diff
@@ -64,7 +64,7 @@ import static org.gradle.internal.logging.text.StyledTextOutput.Style
 // Check for Java 8
 if(JavaVersion.current() != JavaVersion.VERSION_1_8){
     def out = services.get(StyledTextOutputFactory).create("an-ouput")
-    out.withStyle(Style.FailureHeader).println("WARNING: You're compiling with a JDK newer than Java 8, this may not be fully supported for Terasology, consider adjusting your JAVA_HOME accordingly")
+    out.withStyle(Style.FailureHeader).println("NOTE: You're compiling with a JDK newer than 8. While we believe Java 9-11 to be safe to use, newer versions than that may cause issues. If you encounter any oddities try to see if they persist with Java 8")
```

Comment:
Yeah, we're setting that compatibility target in all the java projects. Sometimes through indirect reference to `common.gradle`, but it should still be true.

The one documented case of a java 11 failure we've found is that thing we had with Ubuntu's openjdk distribution the other day. If we do keep a warning like this (and I'm not sure we should; it will be a false positive for the many working java 11 builds), it should have a link to our [troubleshooting page](https://github.com/MovingBlocks/Terasology/wiki/Troubleshooting-Developer#java).

Reply:
Updated 👍 Screenie on Discord. Went with a Java > 11 warning